message.go

```diff
@@ -717,6 +727,35 @@ func (m *RedisMessage) AsZScores() ([]ZScore, error) {
 	return scores, nil
 }
 
+// ScanEntry is the element type of both SCAN, SSCAN, HSCAN and ZSCAN command response.
+type ScanEntry struct {
+	Cursor   int64
+	Elements []string
+}
+
+// AsScanEntry check if message is a redis array/set response of length 2, and convert to ScanEntry.
+func (m *RedisMessage) AsScanEntry() (ScanEntry, error) {
+	msgs, err := m.ToArray()
+	if err != nil {
+		return ScanEntry{}, err
+	}
+	if len(msgs) != 2 {
```

Comment:
Hi @ken8203,

Thanks for the contribution.

```suggestion
	if len(msgs) < 2 {
```

I think it it better to relax this length check. It is not uncommon for redis adding new element to the array in newer version.

Reply:
Sure, it makes sense.